backend-agent/main.py

```diff
@@ -1,8 +1,9 @@
+from argparse import Namespace
 import json
 import os
 from importlib.metadata import version
 
-
+from cli import run_all_logic
```

Comment:
not very elegant, importing the cli module here. Maybe it's better to import the AttackSuite and do the same 3 operations you do in cli (load spec file, set target, and run the suite) even if it would be duplicated.